### PR TITLE
fix: throw error when alias method is used on inkoved solids

### LINKED ISSUE #2670: throw useful error when alias is invoked on solid output
Currently If someone accidentally does

```
@pipeline
def example():
  my_solid().alias('rename_my_solid') # should be my_solid.alias('rename_my_solid')()
```

They get an error message that is pretty difficult to decipher. Since we control the object returned by solid invocation in composition functions, we could override `alias` and throw a good error message to help people find the issue.

### PR DESCRIPTION
**Pull request by iKintosh**
fix #2670

### CODE REVIEW

File: python_modules/dagster/dagster/core/definitions/composition.py [hunk 1]
```diff
@@ -334,6 +334,21 @@ def __getitem__(self, idx):
             )
         )
 
+    def __getattr__(self, attr_called):
+        if attr_called == 'alias':
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/definitions/composition.py:338]: I think better to just define `def alias(self, _):` , that way we don't swallow errors for invocations of other non-existent methods

File: python_modules/dagster/dagster/core/definitions/composition.py [hunk 2]
```diff
@@ -334,6 +334,21 @@ def __getitem__(self, idx):
...
+            raise DagsterInvalidDefinitionError(
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/definitions/composition.py:339]: `DagsterInvariantViolationError`

File: python_modules/dagster/dagster/core/definitions/composition.py [hunk 3]
```diff
@@ -334,6 +334,21 @@ def __getitem__(self, idx):
...
+                'In {source} {name}, attempted to call {attr_name} method for {cls}. This object '
+                'represents the output "{out}" from the already invoked solid "{solid}". Did you '
+                'forget to move parentheses?'.format(
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/definitions/composition.py:342]: > Did you forget to move parentheses?

this statement reads a bit odd to me, I think a statement about how to remedy as opposed to a question assuming past intent is more clear,  "Double check the location of parentheses" or something like that

Comment (iKintosh) [reply python_modules/dagster/dagster/core/definitions/composition.py]: Done.

**Comment by alangenfeld** [comment]
nice, will merge once it passes buildkite

**Pull request merged**